GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/CarStorePageController.cs

```diff
@@ -0,0 +1,138 @@
+﻿using UnityEngine;
+using UnityEngine.UI;
+
+// controller for car store page
+public class CarStorePageController : MonoBehaviour
+{
+    public GameObject itemSedan;
+    public GameObject itemTruck;
+    public GameObject itemJeep;
+    public GameObject itemKart;
+    public GameObject confirmPanel;
+    public Text confirmText;
+
+    private int _priceInCoin;
+    private const int PriceInDollar = -1;
+    private string _carNameToPurchase;
+    private GameManager _gameManager;
+    private GameData _gameData;
+    private readonly Color32 _lightGreyColor = new Color32(147, 147, 147, 255);
+
+    // Start is called before the first frame update
+    private void Start()
+    {
+        _gameManager = FindObjectOfType<GameManager>();
+        _gameData = _gameManager.GetGameData();
+        RefreshPage();
+    }
+    
+    private void OnEnable()
+    {
+        confirmPanel.SetActive(false);
+    }
+    
+    // refresh the page
+    private void RefreshPage()
+    {
+        confirmPanel.SetActive(false);
+        // check if player already owns the car
+        CheckCarOwnership("carSedan", itemSedan, true);
+        CheckCarOwnership("carTruck", itemTruck, false);
+        CheckCarOwnership("carJeep", itemJeep, true);
+        CheckCarOwnership("carKart", itemKart, true);
```

Comment:
Use the public objects you defined in CarList, you can put all cars of CarList into a set there and iterate the set.